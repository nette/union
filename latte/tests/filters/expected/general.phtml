<?php
%A%
final class Template%a% extends Latte\Runtime\Template
{

	public function main(): array
	{
%A%
		echo '<ul>
	<li>';
		echo LR\Filters::escapeHtmlText(($this->filters->h2)(($this->filters->h1)($hello))) /* line %d% */;
		echo '</li>
	<li>';
		echo ($this->filters->h2)(($this->filters->h1)($hello)) /* line %d% */;
		echo '</li>
	<li>';
		echo LR\Filters::escapeHtmlText(($this->filters->h1)(($this->filters->h2)($hello))) /* line %d% */;
		echo '</li>
</ul>

<ul>
	<li>';
		echo LR\Filters::escapeHtmlText(($this->filters->types)((int)$hello*0, 0, 0.0, "0")) /* line %d% */;
		echo '</li>
	<li>';
		echo LR\Filters::escapeHtmlText(($this->filters->types)((int)$hello*1, 1, "1")) /* line %d% */;
		echo '</li>
	<li>';
		echo LR\Filters::escapeHtmlText(($this->filters->types)($hello, true, null, false)) /* line %d% */;
		echo '</li>
	<li>';
		echo LR\Filters::escapeHtmlText(($this->filters->types)($hello, TRUE, NULL, FALSE)) /* line %d% */;
		echo '</li>
	<li>';
		echo LR\Filters::escapeHtmlText(($this->filters->types)($hello, '', "", "$hello")) /* line %d% */;
		echo '</li>
</ul>



';
		ob_start(function () {}) /* line %d% */;
		try {
			echo '  <a   href="#"
> test</a

>
A   A

<script>
// comment
alert();
</script>
';
		} finally {
			$ʟ_fi = new LR\FilterInfo('html');
			echo LR\Filters::convertTo($ʟ_fi, 'html', $this->filters->filterContent('strip', $ʟ_fi, ob_get_clean()));
		}
		echo '



<p>
Nested blocks: ';
		ob_start(function () {}) /* line %d% */;
		try {
			echo ' Outer   ';
			ob_start(function () {}) /* line %d% */;
			try {
				echo ' Inner Block ';
			} finally {
				$ʟ_fi = new LR\FilterInfo('html');
				echo LR\Filters::convertTo($ʟ_fi, 'html', $this->filters->filterContent('upper', $ʟ_fi, $this->filters->filterContent('striphtml', $ʟ_fi, ob_get_clean())));
			}
			echo '  Block ';
		} finally {
			$ʟ_fi = new LR\FilterInfo('html');
			echo LR\Filters::convertTo($ʟ_fi, 'html', $this->filters->filterContent('truncate', $ʟ_fi, $this->filters->filterContent('striphtml', $ʟ_fi, ob_get_clean()), 20));
		}
		echo '
</p>

Breaklines: ';
		echo LR\Filters::escapeHtmlText(($this->filters->breaklines)("hello\nbar")) /* line %d% */;
		echo "\n";
%A%
	}

}
