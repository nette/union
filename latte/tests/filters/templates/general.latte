<ul>
	<li>{$hello |h1|h2}</li>
	<li>{$hello |h1|h2|noescape}</li>
	<li>{$hello |h2|h1}</li>
</ul>

<ul>
	<li>{(int)$hello*0 |types:0,0.0,"0"}</li>
	<li>{(int)$hello*1 |types:1,"1"}</li>
	<li>{$hello |types:true,null,false}</li>
	<li>{$hello |types:TRUE,NULL,FALSE}</li>
	<li>{$hello |types:'',"","$hello"}</li>
</ul>



{block  |strip}  <a   href="#"
> test</a

>
A Â  A

<script>
// comment
alert();
</script>
{/block}



<p>
Nested blocks: {block|striphtml|truncate:20} Outer   {block|striphtml|upper} Inner Block {/block}  Block {/block}
</p>

Breaklines: {="hello\nbar"|breakLines}
