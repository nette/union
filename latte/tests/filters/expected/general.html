<ul>
	<li>HELLO WORLD (1 TIMES)</li>
	<li>HELLO WORLD (2 TIMES)</li>
	<li>hello world (3 times)</li>
</ul>

<ul>
	<li>integer, integer, double, string</li>
	<li>integer, integer, string</li>
	<li>string, boolean, NULL, boolean</li>
	<li>string, boolean, NULL, boolean</li>
	<li>string, string, string, string</li>
</ul>



<a href="#" > test</a> A   A <script>
// comment
alert();
</script>


<p>
Nested blocks:  Outer    INNER…
</p>

Breaklines: hello<br>
bar
